x/mock/types.go

```diff
@@ -0,0 +1,58 @@
+package mock
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+type (
+	// TestAndRunTx produces a random transaction, and check the state
+	// transition. It returns a descriptive message "action" about what this
+	// random tx actually did, for ease of debugging.
+	TestAndRunTx func(
+		t *testing.T, r *rand.Rand, app *App, ctx sdk.Context,
+		privKeys []crypto.PrivKey, log string,
+	) (action string, err sdk.Error)
```

Comment:
It seems like _this_ should be broken down into a "Test and Run 1-Tx" as oppose to choosing a  random Tx for the module - I think the mock suite should probably execute the "randomness" logic to choose which Tx to run, this way we wouldn't have to redefine this logic everywhere - Also what if I want to mix-n-match a few transactions from a few different modules?